Changelog 131 (#93)
intellij-rust/intellij-rust.github.io: _posts/2020-09-21-changelog-131.markdown

---
layout: post
title: "IntelliJ Rust Changelog #131"
date: 2020-09-17 12:00:00 +0300
---

## New Features

<!-- https://github.com/intellij-rust/intellij-rust/pull/6039 -->
* Provide intention to create a method from an unresolved method call (by [@Kobzol])

>     {% include gif-img.html path="/assets/posts/changelog-131/create_method" w="650px" %}

Reviewer: Looks like it's the same gif as in PR. Unfortunately, it's quite blurry. Let's create a new one

<!-- https://github.com/intellij-rust/intellij-rust/pull/5928 -->
* Add initial parsing of [the new asm! macro](https://github.com/rust-lang/rfcs/pull/2873) (by [@Kobzol])

<!-- https://github.com/intellij-rust/intellij-rust/pull/5939 -->
<!-- * RUN: Add build task for wasm-pack configurations -->

<!-- https://github.com/intellij-rust/intellij-rust/pull/5720 -->
* Refactor checking and filling of match expressions

<!-- https://github.com/intellij-rust/intellij-rust/pull/5369 -->

> * Deprioritize completion of `&mut self` methods for const references and bindings (by [@Kobzol])

Reviewer: Maybe something like "Place methods with `&mut self` at the end of completion list for non-mutable variables"?
Just to make it clearer

* Add `dbgr` postfix completion (by [@mgrachev])

    {% include gif-img.html path="/assets/posts/changelog-131/dbgr" w="406px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/6010 -->
* Make moved items public if needed in move refactoring (`Refactor | Move` or <kbd>F6</kbd>)

    {% include gif-img.html path="/assets/posts/changelog-131/move_make_items_public" w="373px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/6057 -->
* Introduce a quick fix to add `..` to struct pattern that has missing fields (by [@t-kameyama])

    {% include gif-img.html path="/assets/posts/changelog-131/pat_rest" w="400px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/5995 -->
* Rename "Cargo Command" run configuration to "Cargo"

    <img src="/assets/posts/changelog-131/cargo_run_configuration_name.png" alt="Cargo run configuration screenshot" width="648px"/>

## Fixes

<!-- https://github.com/intellij-rust/intellij-rust/pull/6075 -->
* Fix "File is not included in module tree" warnings for a project under a symlinked directory

<!-- https://github.com/intellij-rust/intellij-rust/pull/6033 -->

> * Don't resolve std macros when [`no_std`](https://rust-embedded.github.io/book/intro/no-std.html) is present (by [@Kobzol])

Reviewer: ```suggestion
* Don't resolve macros from `std` with `no_std` attribute (by [@Kobzol])
```

<!-- https://github.com/intellij-rust/intellij-rust/pull/6041 -->
* Fix highlighting inside macro calls in some cases

<!-- https://github.com/intellij-rust/intellij-rust/pull/6049 -->
* Fix flattening nested `use` statements intention (by [@t-kameyama])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6078 -->

> * Improve visibility check in move items refactoring

Reviewer: Maybe it's worth to move this item into features section and create top-level item about `Move` refactoring?

<!-- https://github.com/intellij-rust/intellij-rust/pull/6056 -->
* Fix benches flag in `cargo` command completion

<!-- https://github.com/intellij-rust/intellij-rust/pull/6102 -->
<!-- * INT: add missing title and fix typo in CreateFunctionIntention (by [@Kobzol]) -->

## Internal Improvements

<!-- https://github.com/intellij-rust/intellij-rust/pull/6080 -->
* Use the same processor for move files and move items refactoring. It allows adding features to both refactorings simultaneously

<!-- https://github.com/intellij-rust/intellij-rust/pull/6069 -->

> * Cache evcxr executable in CI (by [@Kobzol])

Reviewer: ```suggestion
* Reduce duration of CI checks on Linux by caching `evcxr` executable (by [@Kobzol])
```

<!-- https://github.com/intellij-rust/intellij-rust/pull/6051 -->
* Set pull request milestone automatically

<!-- https://github.com/intellij-rust/intellij-rust/pull/6047 -->
<!-- * TY: relax `expectedType` fallback -->

<!-- https://github.com/intellij-rust/intellij-rust/pull/6042 -->

> * Add test that intentions have descriptions and templates (by [@Kobzol])

Reviewer: ```suggestion
* Check that intentions have description and example (by [@Kobzol])
```

---
layout: post
title: "IntelliJ Rust Changelog #131"
date: 2020-09-21 12:00:00 +0300
---

> 

Reviewer: Since we want to support 2020.3 EAPs and drop 2020.1 support, let's add the corresponding note here.
See `123` release as an example

## New Features

<!-- https://github.com/intellij-rust/intellij-rust/pull/6039 -->
* Provide intention to create a method from an unresolved method call (by [@Kobzol])

>     {% include gif-img.html path="/assets/posts/changelog-131/create_method" w="364px" %}

Reviewer: Maybe make all screenshots bigger (and set the same width). We usually use `700px`. What do you think?

Author: Thanks, good idea

    * Support finding visibility conflicts in more cases

<!-- https://github.com/intellij-rust/intellij-rust/pull/6057 -->
* Introduce a quick fix to add `..` to struct pattern that has missing fields (by [@t-kameyama])

    {% include gif-img.html path="/assets/posts/changelog-131/pat_rest" w="400px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/5995 -->
* Rename "Cargo Command" run configuration to "Cargo"

## Fixes

<!-- https://github.com/intellij-rust/intellij-rust/pull/6075 -->
* Fix "File is not included in module tree" warnings for a project under a symlinked directory

<!-- https://github.com/intellij-rust/intellij-rust/pull/6033 -->
* Don't resolve macros from `std` with `no_std` attribute (by [@Kobzol])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6041 -->
* Fix highlighting inside macro calls in some cases

<!-- https://github.com/intellij-rust/intellij-rust/pull/6049 -->
* Fix flattening nested `use` statements intention (by [@t-kameyama])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6056 -->
* Fix benches flag in `cargo` command completion

<!-- https://github.com/intellij-rust/intellij-rust/pull/6102 -->

> <!-- * INT: add missing title and fix typo in CreateFunctionIntention (by [@Kobzol]) -->

Reviewer: Let's drop commented items - I dropped them from this milestone

Author: I dropped commented items not included in the milestone, but this commit is included in the milestone

Reviewer: Oh, we usually attach all links to the corresponding item
Something like
```
<!-- https://github.com/intellij-rust/intellij-rust/pull/N1 -->
<!-- https://github.com/intellij-rust/intellij-rust/pull/N2 -->
* Some useful text
```
See `130` changelog as an example

Author: Thanks, I forgot about this
